### PR TITLE
Technical/nijo 758 additional refactoring

### PR DESCRIPTION
**Pull request by Raibo**
Issues to fix:
ResultTransactionExtensions.cs
self instead of target
ConfigureAwait is pointless in .Net Core

BookingRecordsManager.cs
new BookingBuilder() looks confusing. Better use static factory if possible

AccountPaymentService.cs 
May it be SingleOrDefault? (316)
Awkward formatting (331)
Why getting booking from context and not from BookingRecordsManager? (316)

### CODE REVIEW

File: Api/Extensions/BookingAddDetailsExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Linq;
+using HappyTravel.Edo.Data.Booking;
+using HappyTravel.EdoContracts.Accommodations;
+using HappyTravel.Money.Models;
+
+namespace HappyTravel.Edo.Api.Extensions
+{
+    public static class BookingAddDetailsExtensions
```
Comment (kirillta) [inline Api/Extensions/BookingAddDetailsExtensions.cs:9]: `BookingExtensions` or `BookingDetailsExtensions`

Comment (mingazhev) [inline Api/Extensions/BookingAddDetailsExtensions.cs:9]: I think just "BookingExtensions" will be better

File: Api/Services/Accommodations/Bookings/BookingFactory.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+using HappyTravel.EdoContracts.General;
+using HappyTravel.EdoContracts.General.Enums;
+using HappyTravel.Money.Models;
+using Newtonsoft.Json;
+
+namespace HappyTravel.Edo.Api.Services.Accommodations.Bookings
+{
+    public static class BookingFactory
+    {
+        public static Booking Build(
```
Comment (mingazhev) [inline Api/Services/Accommodations/Bookings/BookingFactory.cs:19]: Please rename this to something like "Create", "Build" looks awkward for factory methods

File: Api/Services/Accommodations/Bookings/BookingFactory.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                var price = availabilityInfo.RoomContractSet.Price;
+                booking.TotalPrice = price.NetTotal;
+                booking.Currency = price.Currency;
+                booking.Location = new AccommodationLocation(availabilityInfo.CountryName,
+                    availabilityInfo.LocalityName,
+                    availabilityInfo.ZoneName,
+                    availabilityInfo.Address,
+                    availabilityInfo.Coordinates);
+
+                // TODO: Remove code duplication with 'AddBookingDetails'
```
Comment (kirillta) [inline Api/Services/Accommodations/Bookings/BookingFactory.cs:70]: Isn't the time to remove has come?

File: Api/Services/Accommodations/Bookings/BookingRecordsManager.cs [hunk 4]
```diff
@@ -74,18 +75,18 @@ async Task<(string itn, string referenceCode)> GetTags()
...
-                    .AddTags(tags.itn, tags.referenceCode)
-                    .AddStatus(BookingStatusCodes.InternalProcessing)
-                    .AddServiceDetails(availabilityInfo)
-                    .AddPaymentMethod(bookingRequest.PaymentMethod)
-                    .AddRequestInfo(bookingRequest)
-                    .AddLanguageCode(languageCode)
-                    .AddProviderInfo(availabilityInfo.DataProvider)
-                    .AddPaymentStatus(BookingPaymentStatuses.NotPaid)
-                    .Build();
+                var createdBooking = BookingFactory.Build(
```
Comment (kirillta) [inline Api/Services/Accommodations/Bookings/BookingRecordsManager.cs:78]: May we add a new method like `BuildInitial` where `BookingStatusCodes.InternalProcessing` and `BookingPaymentStatuses.NotPaid` will be incapsulated?

Comment (kirillta) [reply Api/Services/Accommodations/Bookings/BookingRecordsManager.cs:78]: And the creation date

**Review by mingazhev — APPROVED** [review]

**Review by kirillta — APPROVED** [review]

**Pull request merged**